src/webgpu/api/operation/adapter/requestDevice.spec.ts

```diff
@@ -78,6 +78,88 @@ g.test('invalid')
     t.expect(lost.reason === undefined);
   });
 
+g.test('stale')
+  .desc(
+    `
+    Test that adapter.requestDevice() can successfully return a device once, and once only.
+    - Tests that we can successfully resolve after serial and concurrent rejections.
+    - Tests that consecutive valid attempts only succeeds the first time, returning lost device otherwise.`
+  )
+  .paramsSubcasesOnly(u =>
+    u
+      .combine('initialError', [undefined, 'TypeError', 'OperationError'])
+      .combine('awaitInitialError', [true, false])
+      .combine('awaitSuccess', [true, false])
+      .filter(({ initialError, awaitInitialError }) => {
+        return initialError !== undefined && !awaitInitialError;
+      })
+  )
+  .fn(async t => {
+    const gpu = getGPU();
+    const adapter = await gpu.requestAdapter();
+    assert(adapter !== null);
+
+    const { initialError, awaitInitialError, awaitSuccess } = t.params;
+
+    switch (initialError) {
+      case undefined:
+        break;
+      case 'TypeError':
+        // Cause a type error by requesting with an unknown feature.
+        if (awaitInitialError) {
+          await assertReject(
+            adapter.requestDevice({ requiredFeatures: ['unknown-feature' as GPUFeatureName] })
+          );
+        } else {
+          t.shouldReject(
+            'TypeError',
+            adapter.requestDevice({ requiredFeatures: ['unknown-feature' as GPUFeatureName] })
+          );
+        }
+        break;
+      case 'OperationError':
+        // Cause an operation error by requesting with an unknown limit.
```

Comment:
An unknown limit is likely to be detected in the frontend (Blink), despite the spec. I think what we really want is something that will fail in the backend (or maybe both). Like, say, setting a limit too high for the adapter, or an unaligned alignment limit.

Reply:
Done.